src/stores/Workflow/Hydrator.tsx

```diff
@@ -1,28 +1,49 @@
-import type { BaseComponentProps } from 'src/types';
+import type { WorkflowInitializerProps } from 'src/components/shared/Entity/types';
 
-import { Fragment } from 'react';
+import { useEffectOnce } from 'react-use';
 
-import DraftInitializer from 'src/components/shared/Entity/Edit/DraftInitializer';
-import { useEntityWorkflow_Editing } from 'src/context/Workflow';
+import Error from 'src/components/shared/Error';
+import { BASE_ERROR } from 'src/services/supabase';
 import BindingHydrator from 'src/stores/Binding/Hydrator';
-import { DetailsFormHydrator } from 'src/stores/DetailsForm/Hydrator';
-import { EndpointConfigHydrator } from 'src/stores/EndpointConfig/Hydrator';
+import { useWorkflowStore } from 'src/stores/Workflow/Store';
+import { useWorkflowHydrator } from 'src/stores/Workflow/useWorkflowHydrator';
 
 // This hydrator is here without a store so that we can start working on moving a lot of
 //  these separate stores into a single "Workflow" store for Create and Edit.
-function WorkflowHydrator({ children }: BaseComponentProps) {
-    const isEdit = useEntityWorkflow_Editing();
-    const InitializerComponent = isEdit ? DraftInitializer : Fragment;
-
-    return (
-        <InitializerComponent>
-            <DetailsFormHydrator>
-                <EndpointConfigHydrator>
-                    <BindingHydrator>{children}</BindingHydrator>
-                </EndpointConfigHydrator>
-            </DetailsFormHydrator>
-        </InitializerComponent>
-    );
+function WorkflowHydrator({
+    children,
+    expressWorkflow,
+}: WorkflowInitializerProps) {
+    const { hydrateState } = useWorkflowHydrator(expressWorkflow);
+
+    const hydrationError = useWorkflowStore((state) => state.hydrationError);
+    const hydrated = useWorkflowStore((state) => state.hydrated);
+    const setActive = useWorkflowStore((state) => state.setActive);
+
+    useEffectOnce(() => {
+        if (!hydrated) {
+            setActive(true);
+            hydrateState();
+        }
+    });
```

Comment:
It is probably safer to make this a `useEffect` and use a `ref` to control it since the `hydrateState` _could_ change based on other deps.

That way we have a bit more control over everything.

Reply:
Leaving a TODO behind for now because metering this effect is not quite that simple.